spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -117,27 +118,31 @@ public InternalRow[] call(InternalRow args) {
           }
 
           String where = args.isNullAt(4) ? null : args.getString(4);
-
-          action = checkAndApplyFilter(action, where, quotedFullIdentifier);
+          if (where != null) {
+            Expression sparkExpression = filter(where, quotedFullIdentifier);
+            if (sparkExpression == null) {
+              // terminate immediately, since the where statement is evaluated by spark to false.
```

Comment:
I don't think this is the right change. If we get null here, it means we couldn't find the filter expression. We should check for an empty scan instead in `collectResolvedSparkExpression`.

Reply:
hi @aokolnychyi ,
thank you so much for the review. 
now I have implemented one possible [solution](https://github.com/apache/iceberg/pull/6760/commits/b8770b19198a1f8ef59964313043af4f02790f84) .
could you help me review it?
thanks